es-app/src/views/ViewController.cpp

```diff
@@ -499,6 +499,28 @@ void ViewController::reloadAll()
 	updateHelpPrompts();
 }
 
+void ViewController::monitorUIMode()
+{
+	std::string uimode = Settings::getInstance()->getString("UIMode");
+	if (uimode != mCurUIMode) // UIMODE HAS CHANGED
+	{	
+		mCurUIMode = uimode;
+		reloadAll();
+		goToStart();
+	}
+}
+
+std::vector<std::string> ViewController::getUIModes()
+{
+	std::vector<std::string> uimodes = { "Full", "Kiosk" };
```

Comment:
I don't know enough about the future plans for this, but for clarity would this be better off as an enum, so that you don't have to do hardcoded text comparisons for what are well-defined terms up front? It's a minor thing, but worth asking. See also "isUIModeFull".

Reply:
But would that also mean we store the UIMode entry in settings as an enum (int)? I would rather have the settings remain user-readable, than prevent a string comparison.

Reply:
Ah. I suppose that's a good argument for keeping it as strings then.

Would we perhaps declare it in the .h for it to be easy to reference, or at least not inside a function if it's meant to have a meaning outside of it?

Either way is fine, just probing.

Thanks.